WhisperAPI/WhisperAPI/Models/SuggestedDocument.cs

```diff
@@ -24,5 +24,26 @@ public SuggestedDocument(ISearchResultElement searchResultElement)
         public string PrintableUri { get; set; }
 
         public string Summary { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) { return false; }
+            if (ReferenceEquals(this, obj)) { return true; }
+            if (obj.GetType() != this.GetType()) { return false; }
+            return this.Equals((SuggestedDocument)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                return ((this.Title != null ? this.Title.GetHashCode() : 0) * 397) ^ (this.Uri != null ? this.Uri.GetHashCode() : 0);
```

Comment:
This is a pretty funky hash code function. Since Uri is unique per document, you could only hash the Uri.